arango/connection.py

```diff
@@ -104,6 +104,8 @@ def prep_response(self, resp: Response, deserialize: bool = True) -> Response:
             if isinstance(resp.body, dict):
                 resp.error_code = resp.body.get("errorNum")
                 resp.error_message = resp.body.get("errorMessage")
+                if resp.status_code == resp.error_code == 503:
+                    raise ConnectionError  # Fallback to another host
         else:
             resp.body = resp.raw_body
```

Comment:
Just curious, in what situations would we not want to deserialize the response raw body?

Reply:
Good question. Although I can't remember exactly which HTTP API endpoints, some did not return JSON objects but strings instead.

Reply:
Ah good to know, thank you 👍